### PR TITLE
copr: vectorize RTrim

### PR DESCRIPTION
**Pull request by MaiCw4J**
Signed-off-by: mak <610546069@qq.com>

PCP #5751 
###  What have you changed?

Add vectorized rtrim

###  What is the type of the changes?

Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No. It should not change the behavior.

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/57

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 1]
```diff
@@ -49,6 +51,19 @@ pub fn ascii(arg: &Option<Bytes>) -> Result<Option<i64>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn rtrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg.as_ref().map(|bytes| {
+        let pos = bytes.iter().rev().position(|&x| x != SPACE);
```
Comment (rleungx) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: How about using `rposition`?

Comment (MaiCw4J) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: cool, that make it more clearer, thx  ;-)

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 2]
```diff
@@ -49,6 +51,19 @@ pub fn ascii(arg: &Option<Bytes>) -> Result<Option<i64>> {
...
 
+#[rpn_fn]
+#[inline]
+pub fn rtrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg.as_ref().map(|bytes| {
+        let pos = bytes.iter().rposition(|&x| x != SPACE);
+        if let Some(i) = pos {
+            bytes[..bytes.len() - i].to_vec()
+        } else {
+            b"".to_vec()
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Maybe you can simply write `Vec::new()`

**Comment by breeswish** [comment]
Please fix conflicts :)

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/57

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/57

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/57

**Pull request merged**